scripts/check_changeset_completeness.ts

```diff
@@ -21,51 +27,80 @@ const getModifiedPackages = (changedFiles: string[]): Set<string> => {
-    '@aws-amplify/backend-storage',
-  ];
-  const backendName: string = '@aws-amplify/backend';
-  const versionBumpOfWrongKind: string[] = [];
-  let backendMaxVersionType: string = 'none';
-
-  for (const changeset of releasePlan.changesets) {
-    for (const release of changeset.releases) {
-      if (release.name === backendName) {
-        if (
-          release.type === 'major' ||
-          backendMaxVersionType === 'none' ||
-          (backendMaxVersionType === 'patch' && release.type === 'minor')
-        ) {
-          backendMaxVersionType = release.type;
-        }
-      }
-    }
+const versionTypeConverter = (version: VersionType): VersionTypeEnum => {
+  switch (version) {
+    case 'major':
+      return VersionTypeEnum.MAJOR;
+    case 'minor':
+      return VersionTypeEnum.MINOR;
+    case 'patch':
+      return VersionTypeEnum.PATCH;
+    case 'none':
+      return VersionTypeEnum.NONE;
   }
+};
+
+const findEffectiveVersion = (
+  releasePlan: ReleasePlan,
+  packageName: string
+): VersionTypeEnum => {
+  let effectiveVersion: VersionTypeEnum = VersionTypeEnum.NONE;
 
   for (const changeset of releasePlan.changesets) {
     for (const release of changeset.releases) {
-      if (backendDependencies.includes(release.name)) {
-        if (
-          backendMaxVersionType !== release.type &&
-          (release.type === 'major' ||
-            backendMaxVersionType === 'none' ||
-            (backendMaxVersionType === 'patch' && release.type === 'minor'))
-        ) {
-          versionBumpOfWrongKind.push(release.name);
+      if (release.name === packageName) {
+        const releaseVersionType = versionTypeConverter(release.type);
+        if (releaseVersionType > effectiveVersion) {
+          effectiveVersion = releaseVersionType;
         }
       }
     }
   }
+  return effectiveVersion;
+};
+
+const maxVersion = (
+  version1: VersionTypeEnum,
+  version2: VersionTypeEnum,
+  version3: VersionTypeEnum,
+  version4: VersionTypeEnum
+) => {
+  return Math.max(version1, version2, version3, version4);
+};
 
-  if (versionBumpOfWrongKind.length > 0) {
+const checkBackendDependenciesVersion = (releasePlan: ReleasePlan) => {
+  const backendVersion: VersionTypeEnum = findEffectiveVersion(
+    releasePlan,
+    '@aws-amplify/backend'
+  );
+  const backendAuthVersion: VersionTypeEnum = findEffectiveVersion(
+    releasePlan,
+    '@aws-amplify/backend-auth'
+  );
+  const backendDataVersion: VersionTypeEnum = findEffectiveVersion(
+    releasePlan,
+    '@aws-amplify/backend-data'
+  );
+  const backendFunctionVersion: VersionTypeEnum = findEffectiveVersion(
+    releasePlan,
+    '@aws-amplify/backend-function'
+  );
+  const backendStorageVersion: VersionTypeEnum = findEffectiveVersion(
+    releasePlan,
+    '@aws-amplify/backend-storage'
+  );
+
+  if (
+    backendVersion <
+    maxVersion(
+      backendAuthVersion,
+      backendDataVersion,
+      backendFunctionVersion,
+      backendStorageVersion
+    )
```

Comment:
```suggestion
    Math.max(
      backendAuthVersion,
      backendDataVersion,
      backendFunctionVersion,
      backendStorageVersion
    )
```

would this just work?

Reply:
yep, the function is unnecessary